### PR TITLE
Update angular to match iteration template

### PR DESCRIPTION
**Pull request by kklamberty**
This PR represents an attempt to bring together changes from Lab 2 and the Iteration Template so that all of the elements of the course will be nicely aligned and ready to go... in theory.

### CODE REVIEW

**Comment by kklamberty** [comment]
Although the checks fail, it's interesting to see at this point how many e2e tests are actually passing!

**Review by NicMcPhee — APPROVED** [review]
I have no idea if this is "everything", but it works nicely and seems to cover all the major pieces that I can remember.

Thanks to @kklamberty for all that hard work!

File: client/cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export default defineConfig({
+  fileServerFolder: '.',
+  fixturesFolder: 'cypress/fixtures',
+  video: true,
+  videosFolder: 'cypress/videos',
+  screenshotsFolder: 'cypress/screenshots',
+  chromeWebSecurity: true,
+  e2e: {
+    // We've imported your old cypress plugins here.
```
[the remark this answers is not in the document]
Comment (kklamberty) [reply client/cypress.config.ts:11]: It looks like dependabot is on the case... I'm going to take care of a spot that I missed node (that dependabot noticed) and then look at this dependabot PR.

File: .gitignore [hunk 2]
```diff
@@ -164,6 +164,7 @@ $RECYCLE.BIN/
 # Aditional custom ignores
 
 .settings/
+.env
```
[the remark this answers is not in the document]
Comment (kklamberty) [reply .gitignore:167]: This was a difference between the iteration template and the lab that I thought we might need. I thought it better to add it and not encounter any .env files that students might accidentally put on GitHub than to leave it out. But, I'll blame "tired stupor" if it was a mistake :P :)

**Pull request merged**